server/mdm/lifecycle/lifecycle.go

```diff
@@ -224,16 +224,20 @@ func (t *HostLifecycle) deleteDarwin(ctx context.Context, opts HostOptions) erro
 	}
 
 	if dep != nil && dep.DeletedAt == nil {
-		return t.restorePendingDEPHost(ctx, opts.Host, ac)
+		return t.restorePendingDEPHost(ctx, opts.Host, dep.ABMTokenID)
 	}
 
 	// no DEP assignment was found or the DEP assignment was deleted in ABM
 	// so nothing more to do
 	return nil
 }
 
-func (t *HostLifecycle) restorePendingDEPHost(ctx context.Context, host *fleet.Host, appCfg *fleet.AppConfig) error {
-	tmID, err := t.getConfigAppleBMDefaultTeamID(ctx, appCfg)
+func (t *HostLifecycle) restorePendingDEPHost(ctx context.Context, host *fleet.Host, abmTokenID *uint) error {
+	if abmTokenID == nil {
```

Comment:
this field is nullable, but in practice it will never be nil, correct? so it's right to error here?

Reply:
yes, sounds good!